Detectors/GlobalTracking/src/MatchTOF.cxx

```diff
@@ -386,9 +497,143 @@ bool MatchTOF::prepareTracks()
+    std::array<float, 3> globalPos;
+
+    // create working copy of track param
+    timeEst timeInfo;
+    // set
+    timeInfo.setTimeStamp(trcOrig.getTime0());
+    timeInfo.setTimeStampError(trcOrig.getDeltaTBwd());
+    mTracksWork.emplace_back(std::make_pair(trcOrig, timeInfo)); //, mCurrTracksTreeEntry, it);
+    o2::track::TrackLTIntegral intLT0;                           //mTPCTracksWork.back().getLTIntegralOut(); // we get the integrated length from TPC-ITC outward propagation
+    mLTinfos.emplace_back(intLT0);
+    // make a copy of the TPC track that we have to propagate
+    //o2::tpc::TrackTPC* trc = new o2::tpc::TrackTPC(trcTPCOrig); // this would take the TPCout track
+    auto& trc = mTracksWork.back().first; // with this we take the TPC track
+    auto& intLT = mLTinfos.back();        // we get the integrated length
+
+    if (trcOrig.getNClusters() < nclustersMin) {
+      nNotPropagatedToTOF++;
+      continue;
+    }
+
+    if (trc.getPt() < minpt) { // tpc-its track outward propagation did not reach outer ref.radius, skip this track
+      nNotPropagatedToTOF++;
+      continue;
+    }
+
+    //    printf("N clusters = %d\n",trcOrig.getNClusters());
+
+    // propagate to matching Xref
+    trc.getXYZGlo(globalPos);
+    LOG(DEBUG) << "Global coordinates Before propagating to 371 cm: globalPos[0] = " << globalPos[0] << ", globalPos[1] = " << globalPos[1] << ", globalPos[2] = " << globalPos[2];
+    LOG(DEBUG) << "Radius xy Before propagating to 371 cm = " << TMath::Sqrt(globalPos[0] * globalPos[0] + globalPos[1] * globalPos[1]);
+    LOG(DEBUG) << "Radius xyz Before propagating to 371 cm = " << TMath::Sqrt(globalPos[0] * globalPos[0] + globalPos[1] * globalPos[1] + globalPos[2] * globalPos[2]);
+    // the "very rough" propagation worked; now we can propagate considering also the cov matrix
+
+    if (!propagateToRefXWithoutCov(trc, mXRef, 10, bzField)) { // we first propagate to 371 cm without considering the covariance matrix
+      nNotPropagatedToTOF++;
+      continue;
+    }
+
+    if (trc.getX() < o2::globaltracking::MatchTPCITS::XTPCOuterRef - 1.)
+      if (!propagateToRefX(trc, o2::globaltracking::MatchTPCITS::XTPCOuterRef, 10, intLT) || TMath::Abs(trc.getZ()) > Geo::MAXHZTOF) { // we check that the propagation with the cov matrix worked; CHECK: can it happen that it does not if the propagation without the errors succeeded?
+        nNotPropagatedToTOF++;
+        continue;
+      }
+
+    // the "rough" propagation worked; now we can propagate considering also the cov matrix
+    if (!propagateToRefX(trc, mXRef, 2, intLT) || TMath::Abs(trc.getZ()) > Geo::MAXHZTOF) { // we check that the propagation with the cov matrix worked; CHECK: can it happen that it does not if the propagation without the errors succeeded?
+      nNotPropagatedToTOF++;
+      continue;
+    }
+
+    trc.getXYZGlo(globalPos);
```

Comment:
getting XYZGlo implies quite some calculations, which you do only for the debugging. Since in any case the DEBUG is compiled out with our default NDEBUG, perhaps it is better to change DEBUG to INFO and put all under the guards e.g.
``#ifdef _PRODUCE_DEBUG_OUTPUT_`` ?

Reply:
Done at line 590 but not a that line since globalPos is used later
    mTracksSectIndexCache[o2::utils::Angle2Sector(TMath::ATan2(globalPos[1], globalPos[0]))].push_back(it);